pkg/paho-mqtt/include/mqtt.h

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2019  Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+/**
+ * @ingroup     pkg_paho-mqtt
+ * @{
+ *
+ * @file
+ * @brief       Network MQTT interface definitions
+ *
+ * @author      Javier FILEIV <javier.fileiv@gmail.com>
+ */
+#ifndef MQTT_H
+#define MQTT_H
+
+#include <mutex.h>
+#include <xtimer.h>
+#include <thread.h>
+#include <net/sock/tcp.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef MQTT_THREAD_PRIORITY
+#define MQTT_THREAD_PRIORITY            (THREAD_PRIORITY_MAIN - 1)
+#endif
+
+#ifndef MQTT_THREAD_STACKSIZE
+#define MQTT_THREAD_STACKSIZE           (THREAD_STACKSIZE_DEFAULT)
+#endif
+
+#ifndef MQTT_YIELD_POLLING_MS
+#define MQTT_YIELD_POLLING_MS           (30)
+#endif
+
+/**
+ * @brief  struct to get time references within mqtt paho
+ *
+ */
+typedef struct Timer
+{
+    xtimer_ticks64_t set_ticks;             /**< timeout ticks */
+    xtimer_ticks64_t ticks_timeout;         /**< timeout in ticks */
+} Timer;
+
+/**
+ * @brief  Initialize timer struct
+ *
+ * @param timer timer to init
+ */
+void TimerInit(Timer* timer);
+
+/**
+ * @brief  is timer expired?
+ *
+ * @param   timer   timer to check
+ *
+ * @return 1 if timer expired, 0 otherwise
+ */
+char TimerIsExpired(Timer* timer);
+
+/**
+ * @brief  start timer set to milli seconds
+ *
+ * @param   timer   timer to start
+ * @param   msecs   time to set in msecs
+ */
+void TimerCountdownMS(Timer* timer, unsigned int msecs);
+
+/**
+ * @brief  start timer set to seconds
+ *
+ * @param   timer   timer to start
+ * @param   secs   time to set in secs
+ */
+void TimerCountdown(Timer* timer, unsigned int secs);
+
+/**
+ * @brief  Returns millisecs left in timer
+ *
+ * @param   timer   timer to check
+ *
+ * @return  msecs left
+ */
+int TimerLeftMS(Timer* timer);
+
+/**
+ * @brief  Network struct within mqtt paho
+ */
+typedef struct Network
+{
+    sock_tcp_t sock;                                                    /**< socket number */
+    int (*mqttread) (struct Network*, unsigned char*, int, int);        /**< read internal function */
+    int (*mqttwrite) (struct Network*, unsigned char*, int, int);       /**< write internal function */
+} Network;
+
+/**
+ * @brief  Initialize network struct
+ *
+ * @param   n   network struct
+ */
+void NetworkInit(Network* n);
```

Comment:
```suggestion
void NetworkInit(Network *n);
```

In fact, in all function declarations